controller/src/main/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasks.java

```diff
@@ -0,0 +1,272 @@
+    }
+
+    /**
+     *  Create a Key-Value Table.
+     *
+     * @param scope      scope name.
+     * @param kvtName    KVTable name.
+     * @param kvtConfig  KVTable configuration.
+     * @param createTimestamp  KVTable creation timestamp.
+     * @return update status.
+     */
+    public CompletableFuture<CreateKeyValueTableStatus.Status> createKeyValueTable(String scope, String kvtName,
+                                                                                   KeyValueTableConfiguration kvtConfig,
+                                                                                   final long createTimestamp) {
+        final long requestId = requestTracker.getRequestIdFor("createKVTable", scope, kvtName);
+        return RetryHelper.withRetriesAsync(() -> {
+               // 1. check if scope with this name exists...
+               return kvtMetadataStore.checkScopeExists(scope)
+                   .thenCompose(exists -> {
+                        if (!exists) {
+                           return CompletableFuture.completedFuture(CreateKeyValueTableStatus.Status.SCOPE_NOT_FOUND);
+                        }
+                        //2. check state of the KVTable, if found
+                        return Futures.exceptionallyExpecting(kvtMetadataStore.getState(scope, kvtName, true, null, executor),
+                                 e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, KVTableState.UNKNOWN)
+                                    .thenCompose(state -> {
+                                       if (state.equals(KVTableState.UNKNOWN) || state.equals(KVTableState.CREATING)) {
+                                           //3. get a new UUID for the KVTable we will be creating.
+                                           byte[] newUUID = kvtMetadataStore.newScope(scope).newId();
+                                           CreateTableEvent event = new CreateTableEvent(scope, kvtName, kvtConfig.getPartitionCount(),
+                                                        createTimestamp, requestId, BitConverter.readUUID(newUUID, 0));
+                                                //4. Update ScopeTable with entry for this KVT and Publish event for creation
+                                                return Futures.exceptionallyExpecting(addIndexAndSubmitTask(event, () -> kvtMetadataStore.createEntryForKVTable(scope, kvtName, newUUID, executor))
+                                                                .thenCompose(x -> checkDone(() -> isCreated(scope, kvtName, kvtConfig, executor))
+                                                                        .thenCompose(y -> kvtMetadataStore.getConfiguration(scope, kvtName, null, executor)
+                                                                                .thenCompose(cfg -> {
+                                                                                    if (cfg.getPartitionCount() == kvtConfig.getPartitionCount()) {
```

Comment:
do a config.equals so that it is future safe. 
otherwise when more configuration is added, this may become a stale value and the developer may not know to change it here.